backend/src/exercise/http-server.ts

```diff
@@ -26,35 +25,29 @@ export class ExerciseHttpServer {
         // This endpoint is used to determine whether the API itself is running.
         // It should be independent from any other services that may or may not be running.
         // This is used for the Cypress CI.
-        app.get('/api/health', (req, res) => {
-            const response = getHealth();
-            res.statusCode = response.statusCode;
-            res.send(response.body);
-        });
-
-        app.post('/api/exercise', async (req, res) => {
-            const response = await postExercise(databaseService, req.body);
-            res.statusCode = response.statusCode;
-            res.send(response.body);
-        });
-
-        app.get('/api/exercise/:exerciseId', (req, res) => {
-            const response = getExercise(req.params.exerciseId);
-            res.statusCode = response.statusCode;
-            res.send(response.body);
-        });
-
-        app.delete('/api/exercise/:exerciseId', async (req, res) => {
-            const response = await deleteExercise(req.params.exerciseId);
-            res.statusCode = response.statusCode;
-            res.send(response.body);
-        });
-
-        app.get('/api/exercise/:exerciseId/history', async (req, res) => {
-            const response = await getExerciseHistory(req.params.exerciseId);
-            res.statusCode = response.statusCode;
-            res.send(response.body);
-        });
+        secureHttp(app, 'get', '/api/health', () => getHealth());
+
+        secureHttp(app, 'post', '/api/exercise', async (req: Request) =>
+            postExercise(databaseService, req.body)
+        );
```

Comment:
Currently, you replace the whole express API to proxy it. What about only wrapping the handler like this:
```ts
        app.get(
            '/api/health',
            secureHttp((req, res) => {
                const response = getHealth();
                res.statusCode = response.statusCode;
                res.send(response.body);
            })
        );
```